dev-packages/e2e-tests/test-applications/nestjs/tests/errors.test.ts

```diff
@@ -54,31 +54,29 @@ test('Does not send expected exception to Sentry', async ({ baseURL }) => {
   expect(errorEventOccurred).toBe(false);
 });
 
-test('Does not handle expected exception if exception is thrown in module', async ({ baseURL }) => {
-  const errorEventPromise = waitForError('nestjs', event => {
-    return !event.type && event.exception?.values?.[0]?.value === 'Something went wrong in the test module!';
-  });
-
-  const response = await fetch(`${baseURL}/test-module`);
-  expect(response.status).toBe(500); // should be 400
+test('Handles exception correctly and does not send to Sentry if exception is thrown in module', async ({
+  baseURL,
+}) => {
+  let errorEventOccurred = false;
 
-  // should never arrive, but does because the exception is not handled properly
-  const errorEvent = await errorEventPromise;
+  waitForError('nestjs', event => {
+    if (!event.type && event.exception?.values?.[0]?.value === 'Something went wrong in the test module!') {
+      errorEventOccurred = true;
+    }
 
-  expect(errorEvent.exception?.values).toHaveLength(1);
-  expect(errorEvent.exception?.values?.[0]?.value).toBe('Something went wrong in the test module!');
+    return event?.transaction === 'GET /test-module';
+  });
 
-  expect(errorEvent.request).toEqual({
-    method: 'GET',
-    cookies: {},
-    headers: expect.any(Object),
-    url: 'http://localhost:3030/test-module',
+  const transactionEventPromise = waitForTransaction('nestjs', transactionEvent => {
+    return transactionEvent?.transaction === 'GET /test-module';
   });
 
-  expect(errorEvent.transaction).toEqual('GET /test-module');
+  const response = await fetch(`${baseURL}/test-module`);
+  expect(response.status).toBe(400);
 
-  expect(errorEvent.contexts?.trace).toEqual({
-    trace_id: expect.any(String),
-    span_id: expect.any(String),
-  });
+  await transactionEventPromise;
+
+  await new Promise(resolve => setTimeout(resolve, 10000));
+
+  expect(errorEventOccurred).toBe(false);
 });
```

Comment:
h: We need to add more tests and make it more clear what exactly is tested.

We should at least add the following two tests:
- Create an app with a user-defined local error filter (`@UseFilters`) and the `SentryIntegrationModule`. Throw an Error that should trigger the local filter and assert that a) the local error filter is called and b) that the error is NOT sent to Sentry.
- Create an app with a user-defined global error filter (`app.useGlobalFilters()` or `provide: APP_FILTER`) and the `SentryIntegrationModule`. Throw an Error that should bubble up and assert that a) the user-defined global error filter is called and b) that the error is NOT sent to Sentry.

These tests are important to assert that we do not alter application behaviour.

Reply:
Added tests for these cases